proof/refine/ARM/Syscall_R.thy

```diff
@@ -2469,14 +2530,627 @@ lemma ct_active_not_idle'[simp]:
+    apply (rule hoare_vcg_conj_lift)
+     apply (wpsimp wp: hoare_vcg_ex_lift getCTE_wp simp: getSlotCap_def)
+    apply (rule hoare_vcg_conj_lift)
+     apply (wpsimp wp: getCTE_wp simp: getSlotCap_def)
+    apply wpsimp
+   apply (rule hoare_vcg_conj_lift)
+    apply (rule_tac p=parent and idx=idx in deleteObjects_invs'[of False])
+   apply (rule hoare_vcg_conj_lift, wp hoare_drop_imp hoare_vcg_ex_lift)
+    apply (rule hoare_vcg_conj_lift)
+     apply (rule_tac p=parent and idx=idx in deleteObjects_cte_wp_at'[where d=False])
+    apply (rule_tac p=parent and idx=idx in deleteObjects_cte_wp_at'[where d=False])
+   apply (rule hoare_vcg_conj_lift)
+    apply (rule_tac idx=idx in deleteObject_no_overlap[where d=False])
+   apply (rule hoare_vcg_conj_lift)
+    apply (rule_tac p=parent and idx=idx in deleteObjects_invs_derivatives(1)[of False])
+   apply (rule hoare_vcg_conj_lift)
+    apply (rule_tac p=parent and idx=idx in deleteObjects_cte_wp_at'[where d=False])
+   apply wpsimp
+  apply (clarsimp simp: cte_wp_at_ctes_of cong: conj_cong)
+  apply (prop_tac " valid_cap' (capability.UntypedCap False frame pageBits idx) s")
+   apply (clarsimp simp: invs'_def valid_pspace'_def dest!: ctes_of_valid')
+  apply (intro conjI; clarsimp?)
+        apply (clarsimp simp: descendants_range'_def2 elim!: empty_descendants_range_in')
+       apply (clarsimp simp: invs'_def valid_pspace'_def)
+       apply (frule_tac p=parent and P="(=) ctea" in if_unsafe_then_capD'[rotated])
+         apply clarsimp
+        apply (clarsimp simp: cte_wp_at_ctes_of)
+       apply (case_tac ctea; simp)
+       apply (drule_tac x=parent in ex_cte_not_in_untyped_range;
+              clarsimp simp: invs'_def valid_pspace'_def)
+      apply (clarsimp simp: valid_cap'_def max_free_index_def)
+     apply (clarsimp simp: max_free_index_def)
+    apply (clarsimp simp: max_free_index_def minUntypedSizeBits_def pageBits_def is_aligned_def)
+   apply (frule (1) active_ex_cap'[OF _ invs_iflive'])
+   apply (drule_tac cap="cteCap ctea" and p=parent in ex_nonz_cap_to_overlap')
+       apply (clarsimp simp: cte_wp_at_ctes_of isUntypedCap_def)+
+    apply (clarsimp simp: descendants_range'_def2 elim!: empty_descendants_range_in')
+   apply clarsimp
+  apply (frule (1) active_ex_cap'[OF _ invs_iflive'])
+  apply (drule_tac cap="cteCap ctea" and p=parent in ex_nonz_cap_to_overlap')
+      apply (clarsimp simp: cte_wp_at_ctes_of isUntypedCap_def)+
+   apply (clarsimp simp: descendants_range'_def2 elim!: empty_descendants_range_in')
+  apply clarsimp
+  apply (meson is_aligned_no_overflow)
+  done
+
+lemma arch_performInv_ct_active':
+  "\<lbrace>invs' and valid_arch_inv' ai and ct_active' and (\<lambda>s. ksSchedulerAction s = ResumeCurrentThread)\<rbrace>
+     ARM_H.performInvocation ai
+   \<lbrace>\<lambda>rv. ct_active'\<rbrace>" (is "\<lbrace>?P\<rbrace> _ \<lbrace>_\<rbrace>")
+  unfolding ARM_H.performInvocation_def performARMMMUInvocation_def
+  apply wpsimp
+       defer 4
```

Comment:
Is it possible to use a `match premises/conclusion` here instead of the `defer`s?